app/models/practice_exercise.rb

```diff
@@ -1,6 +1,21 @@
 class PracticeExercise < Exercise
-  def self.that_practice(concept)
+  has_many :exercise_practiced_concepts,
+    class_name: "Exercise::PracticedConcept",
+    foreign_key: :exercise_id,
+    inverse_of: :exercise,
+    dependent: :destroy
+
+  has_many :practices,
+    through: :exercise_practiced_concepts,
+    source: :concept
+
+  def self.with_prerequisite(concept)
```

Comment:
I added this method, as the existing `that_practice` method was slightly confusing now that we actually have practice concepts.